### PR TITLE
Test decimal breakpoints

### PR DESCRIPTION
**Pull request by ubaldot**
1. Added tests related to PR: https://github.com/vim/vim/pull/16011
2. Adding a new test may not be super-easy for new contributors. I added a tip in the README.txt to help new contributors to tests their own tests within a running Vim instance.

### CODE REVIEW

File: src/testdir/test_termdebug.vim [hunk 1]
```diff
@@ -179,6 +179,60 @@ func Test_termdebug_basic()
   %bw!
 endfunc
 
+func Test_termdebug_decimal_breakpoints()
+  let bin_name = 'XTD_basic'
```
Comment (zeertzjq) [inline src/testdir/test_termdebug.vim]: Shouldn't a different `bin_name` be used?

Comment (ubaldot) [reply src/testdir/test_termdebug.vim]: Point. `let `in legacy script define global variables, correct? How do you suggest to solve the issue then? Just by using a different name than `bin_name`?

Comment (zeertzjq) [reply src/testdir/test_termdebug.vim]: No, it sets function local variables in a function. I was suggesting changing `XTD_basic` to something else.

Comment (ubaldot) [reply src/testdir/test_termdebug.vim]: > No, it sets function local variables in a function.

Auch! I am familiar with Vim9 but with the legacy language I have some problem, sorry!
 
Btw, I changed the name of the file.

Comment (chrisbra) [reply src/testdir/test_termdebug.vim]: when you change it here, you should also change it a few lines below, in the `Termdebug ./bin_name`, no?

Comment (ubaldot) [reply src/testdir/test_termdebug.vim]: Usch! Sorry. I also run the test locally and for some reason they passed... fixed now. Please, take a look. :)

**Comment by chrisbra** [comment]
I still see some failures:
```
command line..script /home/runner/work/vim/vim/src/testdir/runtest.vim[613]..function RunTheTest[57]..Test_termdebug_decimal_breakpoints[24]..WaitForAssert[2]..<SNR>4_WaitForCommon[11]..<lambda>52 line 1: Expected '02' but got '2'
```

**Comment by ubaldot** [comment]
Ok! I see the issue. But I am bit puzzled: why the tests passed when I executed them locally?!

File: src/testdir/test_termdebug.vim [hunk 2]
```diff
@@ -179,6 +179,60 @@ func Test_termdebug_basic()
   %bw!
 endfunc
 
+func Test_termdebug_decimal_breakpoints()
+  let bin_name = 'example_file'
```
Comment (zeertzjq) [inline src/testdir/test_termdebug.vim]: This name is a bit too general. I was suggesting something like `XTD_decimal`

**Comment by ubaldot** [comment]
> Ok! I see the issue. But I am bit puzzled: why the tests passed when I executed them locally?!

I was actually running `make test_termdebug.res` instead of `make test_termdebug`.

File: src/testdir/test_termdebug.vim (hunk 2 above)
Comment (ubaldot) [reply src/testdir/test_termdebug.vim]: I shortened the name. Btw, what XTD stands for?

Comment (zeertzjq) [reply src/testdir/test_termdebug.vim]: `X` is the prefix for temporary files for testing. `TD` means "TermDebug"

Comment (zeertzjq) [reply src/testdir/test_termdebug.vim]: Hmm I mean changing `example_file` to `XTD_decimal`, not changing the test name.

Comment (ubaldot) [reply src/testdir/test_termdebug.vim]: Ok, sorry! I changed it now. Thanks for your guidance. :)

**Comment by chrisbra** [comment]
Still some failures:
```
  Expected:
 ['col', [['leaf', 1002], ['leaf', 1001], ['leaf', 1000], ['leaf', 1003]]]
 but got
 ['col', [['leaf', 1004], ['leaf', 1003], ['leaf', 1000], ['leaf', 1005]]]
```
Can you rename the test to `Test_termdebug_decimal_breakpoint()` so that it runs **after** `Test_termdebug_basic()`

**Comment by ubaldot** [comment]
Thanks @chrisbra for your time!
I further unlet the config dict to be more correct at the end of the test.

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**